File: plurkconnection/src/main/java/tw/shounenwind/plurkconnection/Param.java
package tw.shounenwind.plurkconnection;

public class Param {
    public String key;
    public String value;

    public Param(String key, String value) {
        this.key = key;
        this.value = value;
    }

    @Override
    public String toString() {
        return key + ", " + value;
    }
}

File: library/src/main/java/tw/shounenwind/plurkconnection/Param.kt
package tw.shounenwind.plurkconnection

class Param {
    val key: String
    val value: String

    constructor(key: String, value: String) {
        this.key = key
        this.value = value
    }

    constructor(key: String, value: Long) {
        this.key = key
        this.value = value.toString()
    }

    constructor(key: String, value: Int) {
        this.key = key
        this.value = value.toString()
    }

    constructor(key: String, value: Boolean) {
        this.key = key
        this.value = if (value) "true" else "false"
    }

    override fun hashCode(): Int {
        return arrayOf(key, value).contentHashCode()
    }

    override fun toString(): String {
        return "$key, $value"
    }

    override fun equals(other: Any?): Boolean {
        return other is Param && hashCode() == other.hashCode()
    }
}